bokehjs/src/lib/models/glyphs/webgl/types.ts

```diff
@@ -0,0 +1,150 @@
+import {AttributeConfig, Texture2D, Vec2, Vec4} from "regl"
+
+
+interface CommonProps {
+  canvas_size: Vec2
+  pixel_ratio: number
+  antialias: number
+}
+
+interface LineProps {
+  linewidth: number[] | Float32Array
+  line_color: Uint8Array
+  line_join: number[] | Float32Array
+}
+
+interface LinePropsNoJoin {  // Only needed until Markers support line joins.
+  linewidth: number[] | Float32Array
+  line_color: Uint8Array
+}
+
+interface FillProps {
+  fill_color: Uint8Array
+}
+
+interface DashProps {
+  length_so_far: Float32Array
+  dash_tex: Texture2D
+  dash_tex_info: number[]
+  dash_scale: number
+  dash_offset: number
+}
+
+interface HatchProps {
+  hatch_pattern: number[] | Float32Array
+  hatch_scale: number[] | Float32Array
+  hatch_weight: number[] | Float32Array
+  hatch_color: Uint8Array
+}
+
+export interface LineGlyphProps extends CommonProps {
+  line_color: number[]
+  linewidth: number
+  miter_limit: number
+  points: Float32Array
+  nsegments: number
+  line_cap: number
+  line_join: number
+}
+
+export interface LineDashGlyphProps extends LineGlyphProps, DashProps {}
+
+export interface MarkerGlyphProps extends CommonProps, LinePropsNoJoin, FillProps {
+  center: Float32Array
+  nmarkers: number
+  size: number[] | Float32Array
+  angle: number[] | Float32Array
+  show: Uint8Array
+}
+
+export interface RectGlyphProps extends CommonProps, LineProps, FillProps {
+  center: Float32Array
+  nmarkers: number
+  width: Float32Array
+  height: Float32Array
+  angle: number[] | Float32Array
+  show: Uint8Array
+}
+
+export interface RectHatchGlyphProps extends RectGlyphProps, HatchProps {}
+
+
```

Comment:
Once again multiple empty lines. If you see need for clear separation of section of module's code, then a comment would be preferable explaining following section's purpose.

Reply:
I've switched from 2 empty lines to 1, and added some comments where I think it helps.

Reply:
Actually I've been putting double empty lines in many files, so I've reduced them all to single lines.